README.md

```diff
@@ -182,23 +184,25 @@ kubectl create -f rbac.yaml
 kubectl create -f node-feature-discovery-daemonset.json.template
 ```
 
+When the job runs, it contacts the Kubernetes API server to add labels
+to the node to advertise hardware features.
+
 If you have RBAC authorization enabled (as is the default e.g. with clusters initialized with kubeadm) you need to configure the appropriate ClusterRoles, ClusterRoleBindings and a ServiceAccount in order for NFD to create node labels. The provided templates will configure these for you.
 
 When run as a daemonset, nodes are re-labeled at an interval specified using
 the `--sleep-interval` option. In the [template](https://github.com/kubernetes-incubator/node-feature-discovery/blob/master/node-feature-discovery-daemonset.json.template#L38) the default interval is set to 60s
 which is also the default when no `--sleep-interval` is specified.
 
 Feature discovery can alternatively be configured as a one-shot job. There is
-an example script in this repo that demonstrates how to deploy the job to
-unlabeled nodes.
+an example script in this repo that demonstrates how to deploy the job in the cluster.
 
 ```
 ./label-nodes.sh
 ```
 
-The discovery script will launch a job on each unlabeled node in the
-cluster. When the job runs, it contacts the Kubernetes API server to add labels
-to the node to advertise hardware features (initially, from `cpuid`, RDT, p-state and network).
+The label-nodes.sh script tries to launch as many jobs as there is Ready nodes.
+Note that this approach does not guarantee running once on every node.
+For example, if some node is tainted NoSchedule or fails to start a job for some other reason, then some other node will run extra job instance(s) to satisfy the request.
```

Comment:
should this be clarified a bit more, by adding something like
*...and the tainted/failed node does not get labeled.*